### PR TITLE
Feature/secure element lorawan

### LINKED ISSUE #55: ATECC608A-TNGLORA Integration into basic apps
**Summary:** 

Add basic apps support for ATECC608A-TNGLORA.

**Why do we need this?**

To be able to use the board with and without the secure element.

**What is already there? What do you see now?**

A [feature](https://github.com/TheThingsIndustries/st-node/tree/feature/Basic-ATECC608) branch with fully functional secure element test.

**What is missing? What do you want to see?**

Integration into the basic apps.

**How do you propose to implement this?**

Include the microchip cryptolib as a submodule and Add I2C2 support done in the feature branch. 

**Environment:**

Baremetal

**Acceptance Criteria:**

- Doesn't affect the power consumption of the `basic_lorawan` app.
- Be able to configure the system to use a SE or not.
- The apps compile and function with no warnings

**What can you do yourself and what do you need help with?**

ALL

### PR DESCRIPTION
**Pull request by elsalahy**
**Summary:**

Adds a basic example of secure element LoRaWAN integration because I think we need it to be a separate application
Closes #55 

**Changes:**

- Add stack support for the ATECC608A-TNGLORA and relevant TODOs
- Add secure_element_lorawan application and relevant future TODOs

**Notes for Reviewers:**

You need to claim and activate your secure elements to test this application.

**Release Notes: _(optional)_**

...

### CODE REVIEW

**Comment by elsalahy** [comment]
@marnixcro and @azerimaker  in order for you to test this app, you need to use V3, you should be able to claim your nodes using the [QR code](https://www.thethingsindustries.com/docs/devices/device-claiming/claim-devices/).
Let me know and I can assist you with this.

**Review by johanstokking — DISMISSED** [review]
LGTM, skimmed through it

File: Software/app/secure_element_lorawan/lib_dependency/Commissioning.h [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ The LoRaWAN AES128 keys are stored and provisionned on secure-elements.
+
+ This project providdes a software emulated secure-element.
+ The LoRaWAN AES128 keys SHALL be updated under
+ src/peripherals/<secure-element name>-se\se-identity.h file.
+
+ ******************************************************************************
+ ******************************************************************************
+ ******************************************************************************
+ */
```
Comment (marnixcro) [inline Software/app/secure_element_lorawan/lib_dependency/Commissioning.h:50]: I might be wrong, but this warning doesn't apply for this particular app right?

Comment (marnixcro) [reply Software/app/secure_element_lorawan/lib_dependency/Commissioning.h:50]: \+ minor spelling for all Commissionings: provissioned & provides

**Review by marnixcro — APPROVED** [review]
@elsalahy I have some minor fixes you may want to fix, but the app works and all apps build with cmake/cubeide

File: Software/app/secure_element_lorawan/sys_app.c [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+  */
+static void TimestampNow(uint8_t *buff, uint16_t *size);
+
+/**
+  * @brief  it calls UTIL_ADV_TRACE_VSNPRINTF
+  */
+static void tiny_snprintf_like(char *buf, uint32_t maxsize, const char *strFormat, ...);
+
+/**
+  * @brief initialises the system (dbg pins, trace, mbmux, systiemr, LPM, ...)
```
[the remark this answers is not in the document]
Comment (elsalahy) [reply Software/app/secure_element_lorawan/sys_app.c]: Done fixed

File: Software/app/secure_element_lorawan/lib_dependency/Commissioning.h (hunk 1 above)
Comment (elsalahy) [reply Software/app/secure_element_lorawan/lib_dependency/Commissioning.h]: Done, fixed

**Review by marnixcro — APPROVED** [review]

**Pull request merged**